src/main/java/org/elasticsearch/index/translog/ImmutableTranslogReader.java

```diff
@@ -26,38 +26,43 @@
 import java.nio.ByteBuffer;
 
 /**
- * a channel reader which is fixed in length
+ * a translog reader which is fixed in length
  */
-public final class ChannelImmutableReader extends ChannelReader {
+public class ImmutableTranslogReader extends TranslogReader {
 
     private final int totalOperations;
-    private final long length;
+    protected final long length;
 
     /**
      * Create a snapshot of translog file channel. The length parameter should be consistent with totalOperations and point
      * at the end of the last operation in this snapshot.
      */
-    public ChannelImmutableReader(long id, ChannelReference channelReference, long length, int totalOperations) {
-        super(id, channelReference);
+    public ImmutableTranslogReader(long generation, ChannelReference channelReference, long offset, long length, int totalOperations) {
+        super(generation, channelReference, offset);
```

Comment:
Rename long offset arg to firstOperationOffset to match what super stores it as?